### DISCUSSION TITLE
`mise settings unset` does not seem to work

### DISCUSSION THREAD

**Discussion by scop** [jdx/mise#3865] [Troubleshooting and bug reports]
`mise settings unset` does not seem to work.

```shellsession
$ mise settings
Key             Value Source                    
python.compile  true  ~/.config/mise/config.toml
$ mise settings unset python.compile
$ mise settings
Key             Value Source                    
python.compile  true  ~/.config/mise/config.toml
```

`mise doctor` output
<details>
<pre>
version: 2024.12.21 linux-x64 (71ec24b 2024-12-27)
activated: yes
shims_on_path: no

build_info:
  Target: x86_64-unknown-linux-gnu
  Features: DEFAULT, NATIVE_TLS, OPENSSL
  Built: Fri, 27 Dec 2024 20:25:41 +0000
  Rust Version: rustc 1.83.0 (90b35a623 2024-11-26)
  Profile: release

shell:
  /bin/bash
  GNU bash, version 5.2.21(1)-release (x86_64-pc-linux-gnu)
  Copyright (C) 2022 Free Software Foundation, Inc.
  License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>

  This is free software; you are free to change and redistribute it.
  There is NO WARRANTY, to the extent permitted by law.

dirs:
  cache: ~/.cache/mise
  config: ~/.config/mise
  data: ~/.local/share/mise
  shims: ~/.local/share/mise/shims
  state: ~/.local/state/mise

config_files:
  ~/.config/mise/config.toml

ignored_config_files: (none)

backends:
  aqua
  asdf
  cargo
  core
  dotnet
  gem
  go
  npm
  pipx
  spm
  ubi
  vfox

plugins:

toolset:
  aqua:astral-sh/ruff@0.8.4
  aqua:astral-sh/uv@0.5.13
  aqua:cli/cli@2.64.0
  aqua:crate-ci/committed@1.1.5
  aqua:crate-ci/typos@1.28.4
  aqua:dprint/dprint@0.48.0
  aqua:evilmartians/lefthook@1.10.1
  aqua:golangci/golangci-lint@1.62.2
  aqua:goreleaser/goreleaser@2.5.0
  aqua:goreleaser/nfpm@2.41.1
  aqua:gotestyourself/gotestsum@1.12.0
  aqua:koalaman/shellcheck@0.10.0
  aqua:mvdan/sh@3.10.0
  aqua:sigstore/cosign@2.4.1
  core:go@1.23.4
  ubi:aquaproj/aqua@2.40.0
  ubi:jdx/usage@1.7.4
  ubi:sharkdp/hyperfine@1.19.0
  ubi:suzuki-shunsuke/cmdx@1.7.4

path:
  ~/.local/share/mise/installs/lefthook/1.10.1
  ~/.local/share/mise/installs/uv/0.5.13/uv-x86_64-unknown-linux-musl
  ~/.local/share/mise/installs/golangci-lint/1.62.2/golangci-lint-1.62.2-linux-amd64
  ~/.local/share/mise/installs/shfmt/3.10.0
  ~/.local/share/mise/installs/shellcheck/0.10.0/shellcheck-v0.10.0
  ~/.local/share/mise/installs/goreleaser/2.5.0
  ~/.local/share/mise/installs/go/1.23.4/bin
  ~/.local/share/mise/installs/usage/1.7.4/bin
  ~/.local/share/mise/installs/gh/2.64.0/gh_2.64.0_linux_amd64/bin
  ~/.local/share/mise/installs/typos/1.28.4
  ~/.local/share/mise/installs/gotestsum/1.12.0
  ~/.local/share/mise/installs/dprint/0.48.0
  ~/.local/share/mise/installs/ubi-aquaproj-aqua/2.40.0/bin
  ~/.local/share/mise/installs/ubi-suzuki-shunsuke-cmdx/1.7.4/bin
  ~/.local/share/mise/installs/ruff/0.8.4/ruff-x86_64-unknown-linux-musl
  ~/.local/share/mise/installs/committed/1.1.5
  ~/.local/share/mise/installs/hyperfine/1.19.0/bin
  ~/.local/share/mise/installs/cosign/2.4.1
  ~/.local/share/mise/installs/nfpm/2.41.1
  ~/.local/bin
  ~/.cargo/bin
  ~/.shenv/bin
  ~/.pyenv/bin
  ~/.local/lib/node_modules/.bin
  /usr/lib/ccache
  /usr/local/sbin
  /usr/local/bin
  /usr/sbin
  /usr/bin
  /sbin
  /bin
  /usr/games
  /usr/local/games
  /snap/bin
  /snap/bin
  ~/.local/share/JetBrains/Toolbox/scripts

env_vars:
  MISE_USE_TOML=true
  MISE_SHELL=bash

settings:
  python.compile  true ~/.config/mise/config.toml

No problems found
</pre>
</details>

**Comment by roele** [comment]
Looks like a bug. While `set` is handling keys with `.` explicitly, `unset` is not and the key is not found as expected and therefore not removed.

Handling keys with `.` as `set` does should fix the issue.
https://github.com/jdx/mise/compare/main...roele:mise:discussions/3865?expand=1

**Comment by JamBalaya56562** [comment]
**This is fixed.** 👍

Exactly as @roele diagnosed: `unset` wasn't splitting dotted keys the way `set` does, so a nested key like `python.compile` (stored as `[settings.python] compile = true`) never matched and was left in place.

The fix (@roele's branch) landed in #3867 and shipped in **mise 2024.12.23**. `mise settings unset` now descends into the parent table before removing the child key.

Verified on the current release (`2026.7.12`):
```
$ mise settings set python.compile true
$ mise settings unset python.compile
$ mise settings | grep python.compile
(gone — the setting is actually removed)
```
You were on `2024.12.21`, just before the fix — upgrading resolves it. Could you confirm so this can be closed?

*This comment was generated by an AI coding assistant.*
